static/index.js

```diff
@@ -1,9 +1,19 @@
 (function () {
-  var fs = require('fs')
+  var app = require('remote').require('app')
+  var fs = require('fs-plus')
   var path = require('path')
+  var FileSystemBlobStore = require('../src/file-system-blob-store')
+  var NativeCompileCache = require('../src/native-compile-cache')
 
   var loadSettings = null
   var loadSettingsError = null
+  var blobStore = null
+
+  app.on('before-quit', function () {
```

Comment:
We might want to do this a different way, to avoid doing these two synchronous IPC calls to the browser process (`.require` and `.on`) during startup, and also to avoid contentious writes by multiple windows during shutdown. Maybe the browser process could actually control this in [its `before-quit` callback](https://github.com/atom/atom/blob/69dfdd0745209814b41c0c14e01f3d41ad0d27e8/src/browser/atom-application.coffee#L216), telling only one window to save its state, using some synchronous IPC method like [`WebContents::executeJavascript`](http://electron.atom.io/docs/v0.34.0/api/web-contents/#webcontents-executejavascript-code-usergesture).